python312Packages.blis: 0.7.11 -> 1.0.0

dotlambda opened the pull request:
## Description of changes
Diff: https://github.com/explosion/cython-blis/compare/v0.7.11...release-v1.0.0

Changelog: https://github.com/explosion/cython-blis/releases/tag/release-v1.0.0

Version 1.0 depends on numpy 2.0. C.f. https://github.com/NixOS/nixpkgs/pull/327437.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

dotlambda committed:
python312Packages.blis: 0.7.11 -> 1.0.0

Diff: https://github.com/explosion/cython-blis/compare/v0.7.11...release-v1.0.0

Changelog: https://github.com/explosion/cython-blis/releases/tag/release-v1.0.0

doronbehar commented:
Changes look good. Why is this a draft?

dotlambda commented:
> Changes look good. Why is this a draft?

It breaks thinc and we can't depend on `numpy_2` inside python3Packages.

doronbehar commented:
> > Changes look good. Why is this a draft?
> 
> It breaks thinc and we can't depend on `numpy_2` inside python3Packages.

And `thinc` can't depend on `numpy_2`?

SuperSandro2000 commented:
> It breaks thinc 

Then we need an overlay there

> we can't depend on `numpy_2` inside python3Packages.

I think we can since we catch conflicts now. :) Please give it a shoot.

dotlambda commented:
> > It breaks thinc
> 
> Then we need an overlay there

Thinc is a Python package, that won't work.

dotlambda commented:
The package now works with both `numpy` and `numpy_2`.

dotlambda commented on pkgs/development/python-modules/blis/default.nix:
```diff
@@ -51,17 +59,14 @@ buildPythonPackage rec {
   pythonImportsCheck = [ "blis" ];
 
-  passthru = {
-    # Do not update to BLIS 0.9.x until the following issue is resolved:
-    # https://github.com/explosion/thinc/issues/771#issuecomment-1255825935
```
@davidak Can thinc's numpy pin be removed?

dotlambda replied:
Actually upstream removed it in https://github.com/explosion/thinc/commit/efda3b4e8f4bf7d8518d562feeb3321ae4920309 so this should be fine.

SuperSandro2000 reviewed (approved):
LGTM, can be merged of war resolve the question above this

doronbehar commented on pkgs/development/python-modules/blis/default.nix:
```diff
@@ -3,46 +3,54 @@
-    cython_0
+    cython
+    numpy
   ];
 
-  propagatedBuildInputs = [ numpy ];
+  dependencies = [ numpy ];
```
Isn't that a bit weird to put it in both places?

dotlambda replied:
That's what upstream specifies. IIRC numpy is imported in setup.py.

doronbehar replied:
> That's what upstream specifies. IIRC numpy is imported in setup.py.

Hmm if so, the only good reason to still put Numpy also in the native inputs is for cross compilation, which will not work on NixOS with the way upstream uses the native Numpy in their `setup.py`:

https://github.com/explosion/cython-blis/blob/b33c2ed9cf7ed51d999a926c4950dd8dec4d0286/setup.py#L86C39-L86C58

I think that for cross compilation you'd have to patch that line, and for non-cross compilation, you'd get the host's Numpy available anyway, so there is no need for numpy in the `build-system` list IMO.

natsukium replied:
We plan to separate build dependencies and runtime dependencies in the future. https://github.com/NixOS/nixpkgs/issues/272178
So, we need to add numpy to both build-system and dependencies.

doronbehar replied:
> We plan to separate build dependencies and runtime dependencies in the future. #272178 So, we need to add numpy to both build-system and dependencies.

I'm not sure I agree with everything said there. Even if I had agreed, https://github.com/NixOS/nixpkgs/pull/272179 should have been finished before you'd apply that policy upon other packages.

natsukium replied:
The order is reversed.
Without proper separation, we cannot start #272179.

dotlambda replied:
> you'd get the host's Numpy available anyway, so there is no need for numpy in the `build-system` list IMO.

There is because in the future `dependencies` might not be available at build time in order to reduce the number of rebuilds cause by updates, overrides, etc.

doronbehar replied:
> > you'd get the host's Numpy available anyway, so there is no need for numpy in the `build-system` list IMO.
> 
> There is because in the future `dependencies` might not be available at build time in order to reduce the number of rebuilds cause by updates, overrides, etc.

I support this motivation and this goal, but in this case, due to the goal of upstream's `import numpy` in `setup.py`, it is completely incorrect to add `numpy` to `build-system`, but rather you should patch (downstream or upstream) the file to support specifying manually the host machine's Numpy headers.

dotlambda replied:
I agree that cross doesn't work as is. But at least this continues to work if `dependencies ` are unavailable.

doronbehar replied:
> I agree that cross doesn't work as is. But at least this continues to work if `dependencies ` are unavailable.

OK I understand now. Still it would have been nice if you had put a TODO comment or something. Also the intentions of https://github.com/NixOS/nixpkgs/issues/272178 should be communicated in the documentation IMO, even though not everything is implemented yet.

doronbehar commented on pkgs/development/python-modules/blis/default.nix:
```diff
@@ -51,17 +59,14 @@ buildPythonPackage rec {
-    # Do not update to BLIS 0.9.x until the following issue is resolved:
-    # https://github.com/explosion/thinc/issues/771#issuecomment-1255825935
-    skipBulkUpdate = true;
-    updateScript = gitUpdater {
-      rev-prefix = "v";
-      ignoredVersions = "0\.9\..*";
-    };
```
Why not change the `rev-prefix` to `release-`? Are you sure this is not working?

dotlambda replied:
It's working but shouldn't be necessary.

doronbehar replied:
> It's working but shouldn't be necessary.

You mean that `nix-update` works with the current git tags OK, even without this information? From my experience, it doesn't:

https://github.com/Mic92/nix-update/issues/9#issuecomment-1058215873

dotlambda replied:
You're right, I'll add it back.

dotlambda committed:
python312Packages.thinc: 8.2.3 -> 8.3.0

Changelog: https://github.com/explosion/thinc/releases/tag/v8.2.4
           https://github.com/explosion/thinc/releases/tag/v8.2.5
           https://github.com/explosion/thinc/releases/tag/v8.3.0

dotlambda committed:
python311Packages.spacy: 3.7.5 -> 3.7.6

Changelog: https://github.com/explosion/spaCy/releases/tag/release-v3.7.6

dotlambda committed:
python311Packages.spacy: unpin thinc

dotlambda commented:
Result of `nixpkgs-review pr 333221` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python311Packages.private-gpt</li>
    <li>python311Packages.private-gpt.dist</li>
  </ul>
</details>
<details>
  <summary>88 packages built:</summary>
  <ul>
    <li>aider-chat</li>
    <li>aider-chat.dist</li>
    <li>python311Packages.blis</li>
    <li>python311Packages.blis.dist</li>
    <li>python311Packages.fastai</li>
    <li>python311Packages.fastai.dist</li>
    <li>python311Packages.llama-index</li>
    <li>python311Packages.llama-index-agent-openai</li>
    <li>python311Packages.llama-index-agent-openai.dist</li>
    <li>python311Packages.llama-index-cli</li>
    <li>python311Packages.llama-index-cli.dist</li>
    <li>python311Packages.llama-index-core</li>
    <li>python311Packages.llama-index-core.dist</li>
    <li>python311Packages.llama-index-embeddings-gemini</li>
    <li>python311Packages.llama-index-embeddings-gemini.dist</li>
    <li>python311Packages.llama-index-embeddings-google</li>
    <li>python311Packages.llama-index-embeddings-google.dist</li>
    <li>python311Packages.llama-index-embeddings-huggingface</li>
    <li>python311Packages.llama-index-embeddings-huggingface.dist</li>
    <li>python311Packages.llama-index-embeddings-ollama</li>
    <li>python311Packages.llama-index-embeddings-ollama.dist</li>
    <li>python311Packages.llama-index-embeddings-openai</li>
    <li>python311Packages.llama-index-embeddings-openai.dist</li>
    <li>python311Packages.llama-index-graph-stores-nebula</li>
    <li>python311Packages.llama-index-graph-stores-nebula.dist</li>
    <li>python311Packages.llama-index-graph-stores-neo4j</li>
    <li>python311Packages.llama-index-graph-stores-neo4j.dist</li>
    <li>python311Packages.llama-index-graph-stores-neptune</li>
    <li>python311Packages.llama-index-graph-stores-neptune.dist</li>
    <li>python311Packages.llama-index-indices-managed-llama-cloud</li>
    <li>python311Packages.llama-index-indices-managed-llama-cloud.dist</li>
    <li>python311Packages.llama-index-legacy</li>
    <li>python311Packages.llama-index-legacy.dist</li>
    <li>python311Packages.llama-index-llms-ollama</li>
    <li>python311Packages.llama-index-llms-ollama.dist</li>
    <li>python311Packages.llama-index-llms-openai</li>
    <li>python311Packages.llama-index-llms-openai-like</li>
    <li>python311Packages.llama-index-llms-openai-like.dist</li>
    <li>python311Packages.llama-index-llms-openai.dist</li>
    <li>python311Packages.llama-index-multi-modal-llms-openai</li>
    <li>python311Packages.llama-index-multi-modal-llms-openai.dist</li>
    <li>python311Packages.llama-index-program-openai</li>
    <li>python311Packages.llama-index-program-openai.dist</li>
    <li>python311Packages.llama-index-question-gen-openai</li>
    <li>python311Packages.llama-index-question-gen-openai.dist</li>
    <li>python311Packages.llama-index-readers-database</li>
    <li>python311Packages.llama-index-readers-database.dist</li>
    <li>python311Packages.llama-index-readers-file</li>
    <li>python311Packages.llama-index-readers-file.dist</li>
    <li>python311Packages.llama-index-readers-json</li>
    <li>python311Packages.llama-index-readers-json.dist</li>
    <li>python311Packages.llama-index-readers-llama-parse</li>
    <li>python311Packages.llama-index-readers-llama-parse.dist</li>
    <li>python311Packages.llama-index-readers-s3</li>
    <li>python311Packages.llama-index-readers-s3.dist</li>
    <li>python311Packages.llama-index-readers-twitter</li>
    <li>python311Packages.llama-index-readers-twitter.dist</li>
    <li>python311Packages.llama-index-readers-txtai</li>
    <li>python311Packages.llama-index-readers-txtai.dist</li>
    <li>python311Packages.llama-index-readers-weather</li>
    <li>python311Packages.llama-index-readers-weather.dist</li>
    <li>python311Packages.llama-index-vector-stores-chroma</li>
    <li>python311Packages.llama-index-vector-stores-chroma.dist</li>
    <li>python311Packages.llama-index-vector-stores-google</li>
    <li>python311Packages.llama-index-vector-stores-google.dist</li>
    <li>python311Packages.llama-index-vector-stores-postgres</li>
    <li>python311Packages.llama-index-vector-stores-postgres.dist</li>
    <li>python311Packages.llama-index-vector-stores-qdrant</li>
    <li>python311Packages.llama-index-vector-stores-qdrant.dist</li>
    <li>python311Packages.llama-index.dist</li>
    <li>python311Packages.llama-parse</li>
    <li>python311Packages.llama-parse.dist</li>
    <li>python311Packages.spacy</li>
    <li>python311Packages.spacy-lookups-data</li>
    <li>python311Packages.spacy-lookups-data.dist</li>
    <li>python311Packages.spacy-transformers</li>
    <li>python311Packages.spacy-transformers.dist</li>
    <li>python311Packages.spacy.dist</li>
    <li>python311Packages.textacy</li>
    <li>python311Packages.textacy.dist</li>
    <li>python311Packages.textnets</li>
    <li>python311Packages.textnets.dist</li>
    <li>python311Packages.thinc</li>
    <li>python311Packages.thinc.dist</li>
    <li>python312Packages.blis</li>
    <li>python312Packages.blis.dist</li>
    <li>python312Packages.thinc</li>
    <li>python312Packages.thinc.dist</li>
  </ul>
</details>

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/several-comments-about-priorities-and-new-policies-in-the-python-ecosystem/51790/1
